sql/core/src/test/scala/org/apache/spark/sql/execution/ReduceNumShufflePartitionsSuite.scala

```diff
@@ -59,8 +59,9 @@ class ReduceNumShufflePartitionsSuite extends SparkFunSuite with BeforeAndAfterA
       case (bytesByPartitionId, index) =>
         new MapOutputStatistics(index, bytesByPartitionId)
     }
+    val length = mapOutputStatistics.map(_.bytesByPartitionId.length).head
     val estimatedPartitionStartIndices =
-      rule.estimatePartitionStartIndices(mapOutputStatistics)
+      rule.estimatePartitionStartAndEndIndices(mapOutputStatistics).unzip._1
```

Comment:
we also need to check there is no excluded partitions here, e.g. the end index = next start index - 1

Reply:
Do you mean we need check there is excluded partitions? Current the check is already no excluded partitions.